geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java

```diff
@@ -367,12 +370,38 @@ void doClear(RegionEventImpl regionEvent, boolean cacheWrite) {
           releaseLockForClear(regionEvent);
         }
       }
-
     } finally {
       releaseDistributedClearLock(lockName);
+      incClearDuration(System.nanoTime() - clearStartTime);
+    }
+  }
+
+  void incClearCount() {
+    if (partitionedRegion != null && partitionedRegion.getDataStore() != null
+        && partitionedRegion.getDataStore().getAllLocalBucketRegions() != null
+        && partitionedRegion.getDataStore().getAllLocalBucketRegions().size() != 0) {
+      CachePerfStats stats = partitionedRegion.getCachePerfStats();
+      if (stats != null) {
+        logger.info("BR inc PR Region count:" + stats.getClass().getName() + ":"
+            + partitionedRegion.getFullPath(), new Exception());
+        stats.incClearCount();
+      }
     }
   }
 
+  void incClearDuration(long durationNanos) {
+    if (partitionedRegion != null && partitionedRegion.getTotalNumberOfBuckets() != 0) {
```

Comment:
See my other comment about the possibility of hitting the scenario being checked for. Is it actually possible that we get to where this method is called and `getTotalNumberOfBuckets()` is zero? If it is, do we want to throw an exception instead of just quietly not incrementing the stats?

Reply:
Same reason as above, but I wanted to add that this was simpler because in the original PR this call was only happening on the coordinator. Now this can happen on any member so I've changed it to check everything that was being checked for the clear count.